netket/nn/linear.py

```diff
@@ -234,36 +261,19 @@ class DenseSymm(Module):
 
     def setup(self):
         perms = self.permutations()
-
         self.n_symm, self.n_sites = perms.shape
         self.n_hidden = self.features * self.n_symm
 
-        # symmetrization tensor, maps Wsymm to full W
-        # note that this should be a sparse object, once jax supports them
-        def symmetrizer_ijkl(i, j, k, l):
-            jsymm = np.floor_divide(j, self.n_symm)
-            cond_k = k == jnp.asarray(perms)[j % self.n_symm, i]
-            cond_l = l == jsymm
-            return jnp.asarray(jnp.logical_and(cond_k, cond_l), dtype=int)
-
-        self.symmetrizer = jnp.array(
-            np.fromfunction(
-                symmetrizer_ijkl,
-                shape=(self.n_sites, self.n_hidden, self.n_sites, self.features),
-                dtype=int,
-            ),
-            dtype=self.dtype,
-        ).reshape(-1, self.features * self.n_sites)
+        self.symm_cols = jax.device_put(_symmetrizer_col(perms, self.features))
```

Comment:
you cannot use device_put in jitted blocks.
This kind of works now because we don't jit the init of the neural networks but might change in the future.

See https://github.com/google/jax/issues/6040.

I`m not sure what you are trying to achieve, but suspect that just calling jnp.asarray should work. Netket uses only the default Jax device.

Maybe we should consider refining this behavior at some point, but that's beyond this PR